packages/aws-cdk-lib/aws-cognito/lib/user-pool-client.ts

```diff
@@ -619,4 +669,31 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
       throw new ValidationError(`${name}: Must be a duration between ${min.toHumanString()} and ${max.toHumanString()} (inclusive); received ${value.toHumanString()}.`, this);
     }
   }
+
+  private configureAnalytics(analytics?: AnalyticsConfiguration): CfnUserPoolClient.AnalyticsConfigurationProperty | undefined {
+    if (!analytics) return undefined;
+
+    // NOTE: the CloudFormation expects either `ApplicationArn` or all of `ApplicationId`, `ExternalId` and `RoleArn` to be provided. so validate that these are exclusive.
+    if (
+      analytics.application?.attrArn &&
+        (analytics.applicationId || analytics.externalId || analytics.role)
+    ) {
+      throw new ValidationError('Either `application` or all of `applicationId`, `externalId` and `role` must be specified.', this);
+    }
+
+    if (
+      !analytics.application?.attrArn &&
+        (!analytics.applicationId || !analytics.externalId || !analytics.role)
+    ) {
+      throw new ValidationError('Either all of `applicationId`, `externalId` and `role` must be specified or `application` must be specified.', this);
+    }
```

Comment:
```suggestion
    // NOTE: CloudFormation expects either `ApplicationArn` or all of `ApplicationId`, `ExternalId`, and `RoleArn` to be provided.
    if (
      analytics.application &&
        (analytics.applicationId || analytics.externalId || analytics.role)
    ) {
      throw new ValidationError('Either `application` or all of `applicationId`, `externalId` and `role` must be specified.', this);
    }

    if (
      !analytics.application &&
        (!analytics.applicationId || !analytics.externalId || !analytics.role)
    ) {
      throw new ValidationError('Either all of `applicationId`, `externalId` and `role` must be specified or `application` must be specified.', this);
    }
```
Minor comment and condition adjustment.

Reply:
https://github.com/aws/aws-cdk/pull/32862/commits/6229aff97aef15a4af4fb7e87f64cef398840c2f